experimental/devices/ccs811/ccs811.go

```diff
@@ -0,0 +1,361 @@
+5
+HEATER_SUPPLY
+The Heater voltage is not being applied correctly
+*/
+const (
+	WriteRegInvalid SensorErrorID = 0x1
+	ReadRegInvalid  SensorErrorID = 0x2
+	MeasModeInvalid SensorErrorID = 0x4
+	MaxResistance   SensorErrorID = 0x8
+	HeaterFault     SensorErrorID = 0x10
+	HeaterSupply    SensorErrorID = 0x20
+)
+
+// Opts holds the configuration options.
+type Opts struct {
+	Addr               uint16
+	MeasurementMode    byte
+	InterruptWhenReady bool
+	UseThreshold       bool
+}
+
+// DefaultOpts are the safe default options.
+var DefaultOpts = Opts{
+	Addr:               0x5A,
+	MeasurementMode:    MeasurementModeConstant1000,
+	InterruptWhenReady: false,
+	UseThreshold:       false,
+}
+
+// New creates a new driver for CCS811 VOC sensor
+func New(bus i2c.Bus, opts *Opts) (*Dev, error) {
+	if opts.Addr != 0x5A && opts.Addr != 0x5B {
+		return nil, fmt.Errorf("Invalid device address, only 0x5A or 0x5B are allowed")
+	}
+
+	if opts.MeasurementMode > 4 {
+		return nil, fmt.Errorf("Invalid measurement mode")
+	}
+
+	dev := &Dev{
+		c:    &i2c.Dev{Bus: bus, Addr: opts.Addr},
+		opts: opts,
+	}
+
+	// from boot mode to measurement mode
+	err := dev.StartSensorApp()
+	if err != nil {
+		return nil, fmt.Errorf("Error transitioning from boot do app mode: %v", err)
+	}
+
+	time.Sleep(20 * time.Millisecond)
+
+	dev.SetMeasurementMode(MeasurementModeConstant1000, opts.InterruptWhenReady, opts.UseThreshold)
+
+	return dev, nil
+}
+
+// Dev is an handle to an CCS811 sensor.
+type Dev struct {
+	c    conn.Conn
+	opts *Opts
+}
+
+const ( //registers
+	statusReg          byte = 0x00
+	measurementModeReg byte = 0x01
+	algoResultsReg     byte = 0x02
+	rawDataReg         byte = 0x03
+	environmentReg     byte = 0x05
+	baselineReg        byte = 0x11
+	resetReg           byte = 0xFF
+)
+
+func (d *Dev) String() string {
+	return "CCS811"
+}
+
+// StartSensorApp initializes sensor to application mode
+func (d *Dev) StartSensorApp() error {
+	err := d.c.Tx([]byte{0xf4}, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetMeasurementMode sets one of the 5 measurement modes, interrupt generation
+// and interrupt threshold
+//
+// generateInterrupt:
+//		if true, CCS811 will trigger interrupt when new data is available
+// useThreshold:
+// 		if true, you have to set Threshold register with appropriate values
+func (d *Dev) SetMeasurementMode(measurementMode byte, generateInterrupt, useThreshold bool) error {
+	mesModeValue := (measurementMode << 4)
+	if generateInterrupt {
+		mesModeValue = mesModeValue | (0x1 << 3)
+	}
+	if useThreshold {
+		mesModeValue = mesModeValue | (0x1 << 2)
+	}
+
+	// set measurement mode
+	err := d.c.Tx([]byte{measurementModeReg, mesModeValue}, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type MeasurementMode struct {
+	MeasurementMode   byte
+	GenerateInterrupt bool
+	UseThreshold      bool
+}
+
+func (d *Dev) GetMeasurementMode() (mm MeasurementMode, err error) {
```

Comment:
Don't use named return values on all functions. The only exception is when a return value is needed in a defered function.